src/fire_detector/fire_detector.py

```diff
@@ -0,0 +1,76 @@
+import tensorflow as tf
+import cv2
+import numpy as np
+from typing import Tuple
+
+
+class FireDetector:
+    def __init__(self, model_path: str):
+        """
+        Initialize a FireDetector instance with a pre-trained model.
+
+        Args:
+            model_path (str): The path to the pre-trained model.
+        """
+        self.__model = tf.keras.models.load_model(model_path)
+
+    @staticmethod
+    def __resize_and_pad_image(image: np.ndarray, output_size: Tuple[int, int]) -> np.ndarray:
+        """
+        Resize an image to the specified size while maintaining its original aspect ratio.
+        If necessary, pad the image to achieve the exact size.
+
+        Args:
+            image (np.ndarray): The image to resize.
+            output_size (tuple): A tuple (width, height) specifying the desired output size (e.g., (128, 128)).
+
+        Returns:
+            np.ndarray: The resized and padded image as a NumPy array.
+        """
+
+        # Calculate the aspect ratio
+        original_height, original_width, _ = image.shape
+        target_width, target_height = output_size
+        aspect_ratio = original_width / original_height
+
+        # Calculate the new dimensions while maintaining the aspect ratio
```

Comment:
remove redundant documentation inside function's , the code is self explanatory

Reply:
done